substrate/frame/examples/basic/src/lib.rs

```diff
@@ -82,92 +97,39 @@ pub use weights::*;
-//
-// The `WeightData<T>` trait has access to the arguments of the dispatch that it wants to assign a
-// weight to. Nonetheless, the trait itself cannot make any assumptions about what the generic type
-// of the arguments (`T`) is. Based on our needs, we could replace `T` with a more concrete type
-// while implementing the trait. The `pallet::weight` expects whatever implements `WeighData<T>` to
-// replace `T` with a tuple of the dispatch arguments. This is exactly how we will craft the
-// implementation below.
-//
-// The rules of `WeightForSetDummy` are as follows:
-// - The final weight of each dispatch is calculated as the argument of the call multiplied by the
-//   parameter given to the `WeightForSetDummy`'s constructor.
-// - assigns a dispatch class `operational` if the argument of the call is more than 1000.
-//
-// More information can be read at:
-//   - https://docs.substrate.io/main-docs/build/tx-weights-fees/
-//
-// Manually configuring weight is an advanced operation and what you really need may well be
-//   fulfilled by running the benchmarking toolchain. Refer to `benchmarking.rs` file.
-struct WeightForSetDummy<T: pallet_balances::Config>(BalanceOf<T>);
-
-impl<T: pallet_balances::Config> WeighData<(&BalanceOf<T>,)> for WeightForSetDummy<T> {
-	fn weigh_data(&self, target: (&BalanceOf<T>,)) -> Weight {
-		let multiplier = self.0;
-		// *target.0 is the amount passed into the extrinsic
-		let cents = *target.0 / <BalanceOf<T>>::from(MILLICENTS);
-		Weight::from_parts((cents * multiplier).saturated_into::<u64>(), 0)
-	}
-}
-
-impl<T: pallet_balances::Config> ClassifyDispatch<(&BalanceOf<T>,)> for WeightForSetDummy<T> {
-	fn classify_dispatch(&self, target: (&BalanceOf<T>,)) -> DispatchClass {
-		if *target.0 > <BalanceOf<T>>::from(1000u32) {
-			DispatchClass::Operational
-		} else {
-			DispatchClass::Normal
-		}
-	}
-}
 
-impl<T: pallet_balances::Config> PaysFee<(&BalanceOf<T>,)> for WeightForSetDummy<T> {
-	fn pays_fee(&self, _target: (&BalanceOf<T>,)) -> Pays {
-		Pays::Yes
-	}
-}
-
-// Definition of the pallet logic, to be aggregated at runtime definition through
-// `construct_runtime`.
+// Definition of the pallet logic, to be aggregated in a chain's runtime definition.
 #[frame_support::pallet]
 pub mod pallet {
 	// Import various types used to declare pallet in scope.
 	use super::*;
 	use frame_support::pallet_prelude::*;
 	use frame_system::pallet_prelude::*;
 
-	/// Our pallet's configuration trait. All our types and constants go in here. If the
-	/// pallet is dependent on specific other pallets, then their configuration traits
-	/// should be added to our implied traits list.
+	/// Our pallet's configuration trait.
 	///
-	/// `frame_system::Config` should always be included.
+	/// Because all FRAME pallets depend on some core utility types from the System pallet,
+	/// [`frame_system::Config`] should always be included. This pallet example uses the `Balances`
```

Comment:
Things like `Config` have some default doc that is generated by the pallet macro for them as well, FYI. If you provide your own docs, it will be overwritten.